src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs

```diff
@@ -0,0 +1,191 @@
+    [(entity "namespace")
+     (entity "name")
+     (entity "snapshotId")]))
+
+(defn- is-conf? [e]
+  (contains? e "method"))
+
+(def ^:private access-levels
+  ["READER" "OWNER" "NO ACCESS"])
+
+(defn- build-acl-vec [acl-map]
+  (mapv
+    (fn [k] {:userId k :accessLevel (acl-map k)})
+    (keys acl-map)))
+
+
+(defn- get-agora-method-acl [ns n sid is-conf]
+  {:path (str "/" (if is-conf "configurations" "methods"  ) "/" ns "/" n "/" sid "/permissions"  )
+   :method :get
+   :mock-data
+   (map (fn [i] {(str "user" ) (str "user" i)
+                 (str "roles") (rand-nth
+                                 [["Read"]
+                                  ["Read" "Write" "Create" "Redact" "Manage"]
+                                  []])})
+     (range (inc (rand-int 5))))})
+
+(defn- ui-acl-to-agora-perms [ui-acl-str]
+  (cond
+    (= "NO ACCESS" ui-acl-str) [] ; empty array for no access
+    (= "READER" ui-acl-str) ["Read"] ; just reader
+    (= "OWNER" ui-acl-str) ["Read","Write","Create","Redact","Manage"])) ; all 5 perms
+
+(defn- translate-single-acl-map [a-map]
+  (let [ui-role (a-map :accessLevel)
+        agora-arr (ui-acl-to-agora-perms ui-role)
+        new-map (merge a-map {:accessLevel agora-arr})]
+    (utils/rlog (str "the ui-role is " ui-role))
+        new-map))
+
+(defn- agora-perms-to-ui-acl [agora_perm_arr]
+  (let [get-user (fn [e] (e "user"))
+        get-roles (fn [e] (e "roles"))
+        map-role-arr-to-ui-role (fn [role-arr]
+                    (cond
+                      (= 0 (count role-arr))
+                      "NO ACCESS" ; no entries maps to "NO ACCESS"
+                      (and (= 1 (count role-arr)) (= "Read" (nth role-arr 0)))
+                      "READER" ;exactly one entriy 'Read' maps to "READER"
+                      (= 5 (count role-arr))
+                      "OWNER" ; 5 entries maps to "OWNER"
+                      :else "READER"))
+        mapping-func (fn [perm-item]
+                       (let [user (get-user perm-item)
+                             roles (get-roles perm-item)
+                             ui-role (map-role-arr-to-ui-role roles)]
+                         {user ui-role}))]
+    (map mapping-func agora_perm_arr)))
+
+
+(defn- persist-agora-method-acl [ent]
+  {:path (let [ent-type (ent "entityType")
+               name (ent "name")
+               nmsp (ent "namespace")
+               sid (ent "snapshotId")
+               base (cond
+                      (= "Configuration" ent-type) "/configurations"
+                      (or (= "Task" ent-type) (= "Workflow" ent-type)) "methods"
+                      :else (do
+                              (utils/rlog "Error, unknown type : " ent-type)
+                              (str "configurations")
+                              ))]
+           (str "/" base "/" nmsp "/" name "/" sid "/permissions"))
+   :method :post})
+
+(def ^:private column-width "calc(50% - 4px)")
+
+(react/defc AgoraPermsEditor
+  { :persist-acl
```

Comment:
Re: "This and capture-ui-state...."  My understanding here is that you are referring to the ORDER of code only is that correct?  I can cut the functions, move the cursor higher, then paste (making sure that proper indentation (as defined here  https://broadinstitute.atlassian.net/wiki/display/DSDE/Code+Style+Conventions) is achieved?  Would that address the concern here?